framework/src/headers/ip/v6/icmp/ndp.rs

```diff
@@ -0,0 +1,124 @@
+use std::default::Default;
+use std::fmt;
+use std::marker::PhantomData;
+use std::net::Ipv6Addr;
+
+use headers::{CalcChecksums, EndOffset, Ipv6VarHeader};
+
+use super::{IcmpMessageType, Icmpv6Header};
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct NdpMessage<T>
```

Comment:
NdpMessage was removed since it wasn't being used by anything. It was originally part of the NeighborAdvertisement and NeighborSolicitation messages but those were removed from the PR. Paul's PR only contains the RouterAdvertisement since that's all we care about now